### PR TITLE
service CLI improvements

### PR DESCRIPTION
**Pull request by benjigoldberg**
@briancharous I realized I glossed over some of the CLI config for the generic http service. This just adds that back and makes it a bit more common.

### CODE REVIEW

File: service/cli.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"github.com/spf13/pflag"
+)
+
+// Config defines service level configuration for HTTP servers
+type Config struct {
+	Name        string               // Name of the application
+	Version     string               // Semantic Version of the application
+	GitSHA      string               // GitSHA of the application when compiled
+	Environment string               // Environment where the server is running
+	Package     string               // The name of this go package, eg `github.com/spothero/myservice`
```
Comment (briancharous) [inline service/cli.go]: This is used for sentry only right? Not sure if this is possible but can we nix it here and use reflection to get this value elsewhere?

File: service/cli.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	Registry    *prometheus.Registry // The Prometheus Registry to use. If nil, the global registry is used by default.
+}
+
+// RegisterFlags registers Service flags with pflags
+func (c *Config) RegisterFlags(flags *pflag.FlagSet) {
+	flags.StringVarP(&c.Name, "name", "n", c.Name, "Name of the application")
+	flags.StringVarP(&c.Version, "version", "v", c.Version, "Semantic Version of the application")
+	flags.StringVarP(&c.GitSHA, "git-sha", "g", c.GitSHA, "Git SHA of the application")
```
Comment (briancharous) [inline service/cli.go]: Adding Git SHA and Version as cli options is weird - they should be baked in with linker flags.

Comment (benjigoldberg) [reply service/cli.go]: was trying to enforce this, but I agree with the argument.

File: Makefile [hunk 3]
```diff
@@ -10,7 +12,7 @@ tidy:
 	go mod tidy
 
 build: tidy
-	go build -ldflags="-X main.version=${VERSION} -X main.gitSha=${GIT_SHA}" examples/example_server.go
+	go build -ldflags="-X main.version=${VERSION} -X main.gitSHA=${GIT_SHA}" examples/example_server.go
```
Comment (briancharous) [inline Makefile]: remove the clean target since there's no vendoring

File: sentry/sentry.go [hunk 4]
```diff
@@ -42,7 +42,12 @@ func (c Config) InitializeRaven() error {
 	}
 	raven.SetEnvironment(c.Environment)
 	raven.SetRelease(c.AppVersion)
-	appPackage = c.AppPackage
+
+	buildInfo, ok := debug.ReadBuildInfo()
```
Comment (briancharous) [inline sentry/sentry.go]: fancy

**Pull request merged**